### PR TITLE
UX: Apply new quote-button spacing behavior to all mobile devices

### PR DESCRIPTION
**Pull request by davidtaylorhq**
This expands cbf99f48 to apply to all mobile devices. It removes the old mobile positioning logic entirely, refactors the new system a little for robustness and readability, and removes some JQuery.

### CODE REVIEW

**Review by ZogStriP — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/quote-button.js [hunk 1]
```diff
@@ -229,55 +228,77 @@ export default Component.extend(KeyEnterEscape, {
...
-        top = top - $quoteButton.outerHeight() - 5;
-      }
 
-      if (isIOS) {
+      if (showAtEnd) {
         // The selection-handles on iOS have a hit area of ~50px radius
         // so we need to make sure our buttons are outside that radius
+        // Apply the same logic on all mobile devices for consistency
 
+        top = top + 25;
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/quote-button.js:237]:
```suggestion
        top += 25;
```

**Review by CvX — APPROVED** [review]

**Comment by davidtaylorhq** [comment]
(Dismissing reviews since I just force-pushed a substantial refactor so that the 'selection boundary location' logic is re-used)

File: app/assets/javascripts/discourse/app/components/quote-button.js [hunk 2]
```diff
@@ -184,100 +215,98 @@ export default Component.extend(KeyEnterEscape, {
...
-
-    // work around Safari that would sometimes lose the selection
-    if (isSafari) {
-      this._reselected = true;
-      selection.removeAllRanges();
-      selection.addRange(clone);
-    }
+    const boundaryPosition = getRangeBoundaryRect(
+      firstRange,
+      showAtEnd ? true : false
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/quote-button.js:223]:
```suggestion
      showAtEnd
```

Or is one of those helpers returning a non-bool? 🙂

Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/quote-button.js:223]: Fixed :+1:

**Review by CvX — APPROVED** [review]

**Pull request merged**